### PR TITLE
boost-beast: initial integration

### PR DESCRIPTION
**Pull request by tyler92**
This pull request replaces [the previous one](https://github.com/google/oss-fuzz/pull/11994), where targets were stored in the oss-fuzz repository. With the new approach, targets and seeds will be stored in the upstream repository instead.

### CODE REVIEW

File: projects/boost-beast/Dockerfile
Comment (tyler92) [inline projects/boost-beast/Dockerfile:20]: I will remove this line when changes from https://github.com/boostorg/beast/pull/2888 are merged not only into develop, but into master too

Comment (tyler92) [reply projects/boost-beast/Dockerfile:20]: We decided to leave it as is (the same for boost-json project)

**Comment by tyler92** [comment]
Do I need the maintainer's approval first?

**Comment by tyler92** [comment]
@DavidKorczynski @jonathanmetzman Hi! I would appreciate it if you have a look. Sorry to ping you directly, but I don't know how to assign the PR to somebody

**Comment by DonggeLiu** [comment]
Thanks @tyler92 
May I ask who are the major users of the project and why is it critical?
This will help the panel approve this PR.

**Comment by tyler92** [comment]
Hi @DonggeLiu 

Boost.Beast is a part of the popular project Boost and it is critical because it handles user input for HTTP and WebSocket communications, making its security and reliability essential. It's used in web servers, proxies, embedded systems, IoT. Integrating with OSS-Fuzz will help identify and fix vulnerabilities, enhancing its robustness and benefiting many applications and the broader Boost ecosystem. Some time ago, the maintainers expressed a desire to integrate this project: https://github.com/google/oss-fuzz/issues/862

Also, please note that it's not really a new integration. There is already [Boost project](https://github.com/google/oss-fuzz/tree/master/projects/boost) integrated to oss-fuzz that includes different parts of Boost. I can move Beast there, but it uses a slightly different approach - fuzzing targets and seeds are [stored in the upstream repository](https://github.com/boostorg/beast/pull/2888). That's why I decided to make it a separate project here, like with [boost-json](https://github.com/google/oss-fuzz/tree/master/projects/boost-json). However, please let me know if it's better to move Beast to projects/boost and consider this PR not as a new integration, but as an extension of the existing integration.

**Comment by DonggeLiu** [comment]
I am waiting for the panel to decide.

**Review by DonggeLiu — APPROVED** [review]
The panel has approved

**Pull request merged**
